### PR TITLE
Catch IllegalArgumentException in getInputStream

### PR DESCRIPTION
**Pull request by beylmk**
[sc-12011]

Prevents a user-reported crash when there is an exception setting up the network connection (see shortcut ticket for more details)

### CODE REVIEW

File: common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt [hunk 1]
```diff
@@ -58,8 +58,15 @@ class HTTPClient(
     private fun getInputStream(connection: HttpURLConnection): InputStream? {
         return try {
             connection.inputStream
-        } catch (e: IOException) {
-            connection.errorStream
+        } catch(e: Exception) {
```
Comment (NachoSoto) [inline common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt:61]: At least in Java you can do `catch(e: IOException | IllegalArgumentException)`. I wonder if that works here too?

Comment (taquitos) [reply common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt:61]: Seven. Years. Old.
https://youtrack.jetbrains.com/issue/KT-7128

Comment (NachoSoto) [reply common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt:61]: Haha wow

Comment (vegaro) [inline common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt:61]:
```suggestion
        } catch (e: Exception) {
```

**Review by vegaro — APPROVED** [review]

File: common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt (hunk 1 above)
Comment (vegaro) [reply common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt:61]: Detekt complains about catching a too generic Exception, which is fine to ignore. Alternatively, you could do the multiple `catch` version and create a function (I think):

```
try {

catch (e: IllegalArgumentException) {
  handleException(e)
catch (e: IOException) {
  handleException(e)
}
```

**Pull request merged**